test/auth_test.js

```diff
@@ -276,677 +61,837 @@ suite('api/auth', function() {
+  const testEndpoint = ({method, route, scopes = null, handler, tests}) => {
+    api.declare({
+      method,
+      route,
+      name: 'placeholder',
+      title: 'placeholder',
+      description: 'placeholder',
+      scopes,
+    }, handler);
+    const buildUrl = (params = {}) => {
+      const path = route.replace(/:[a-zA-Z][a-zA-Z0-9]+/g, match => {
+        const result = params[match.replace(/^:/, '')];
+        if (!result) {
+          throw new Error('Bad test, must specifiy all route params!');
         }
+        return result;
       });
-  });
-
-  test('scope expression if/then (success with no client) 2', function() {
-    var url = 'http://localhost:23526/test-expression-if-then-2';
-    return request
-      .get(url)
-      .send({
-        public: true,
-      })
-      .then(function(res) {
-        if (!res.ok) {
-          console.log(JSON.stringify(res.body));
-          assert(false, 'Request failed');
-        }
-      });
-  });
-
-  test('scope expression if/then (failure) 2', function() {
-    var url = 'http://localhost:23526/test-expression-if-then-2';
-    return request
-      .get(url)
-      .send({
-        public: false,
-      })
-      .hawk({
-        id:           'rockstar',
-        key:          'groupie',
-        algorithm:    'sha256',
-      }, {
-        ext: new Buffer(JSON.stringify({
-          authorizedScopes:    ['nothing:useful'],
-        })).toString('base64'),
-      })
-      .then(res => assert(false, 'request didn\'t fail'))
-      .catch(function(res) {
-        assert(res.status === 403, 'Request didn\'t fail');
-      });
-  });
-
-  test('scope expression if/then (failure with no client) 2', function() {
-    var url = 'http://localhost:23526/test-expression-if-then-2';
-    return request
-      .get(url)
-      .send({
-        public: false,
-      })
-      .then(res => assert(false, 'request didn\'t fail'))
-      .catch(function(res) {
-        assert(res.status === 403, 'Request didn\'t fail');
-      });
-  });
-
-  test('scope expression if/then (forgot to auth)', function() {
-    var url = 'http://localhost:23526/test-expression-if-then-forget';
-    return request
-      .get(url)
-      .send({})
-      .hawk({
-        id:           'rockstar',
-        key:          'groupie',
-        algorithm:    'sha256',
-      })
-      .then(res => assert(false, 'request didn\'t fail'))
-      .catch(function(res) {
-        assert(res.status === 500, 'Request didn\'t fail');
+      return `http://localhost:23526${path}`;
+    };
+    tests.forEach(({label, params, testfactory}) => test(label, testfactory(buildUrl(params))));
+  };
+
+  testEndpoint({
+    method: 'get',
+    route:  '/test-deprecated-satisfies',
+    handler: (req, res) => {
+      if (req.satisfies([])) {
+        res.status(200).json({ok: true});
+      }
+    },
+    tests: [
+      {
+        label: 'function that still uses satisfies fails',
+        testfactory: url => () => {
```

Comment:
I think you could make this declarative too...

As in list what should be I've to hawk.. what should be expected as result... Status code etc...

And then just have a function as fallback...

Reply:
Also a different name than test factory please...